client/src/components/TimePreferencesModal.js

```diff
@@ -0,0 +1,351 @@
+            value: day
+        }
+    })
+const timeSlotOptions = [
+    {
+        key: 'slot1',
+        value: 0,
+        text: '12am - 1am'
+    },
+    {
+        key: 'slot2',
+        value: 100,
+        text: '1am - 2am'
+    },
+    {
+        key: 'slot3',
+        value: 200,
+        text: '2am - 3am'
+    },
+    {
+        key: 'slot4',
+        value: 300,
+        text: '3am - 4am'
+    },
+    {
+        key: 'slot5',
+        value: 400,
+        text: '4am - 5am'
+    },
+    {
+        key: 'slot6',
+        value: 500,
+        text: '5am - 6am'
+    },
+    {
+        key: 'slot7',
+        value: 600,
+        text: '6am - 7am'
+    },
+    {
+        key: 'slot8',
+        value: 700,
+        text: '7am - 8am'
+    },
+    {
+        key: 'slot9',
+        value: 800,
+        text: '8am - 9am'
+    },
+    {
+        key: 'slot10',
+        value: 900,
+        text: '9am - 10am'
+    },
+    {
+        key: 'slot11',
+        value: 1000,
+        text: '10am - 11am'
+    },
+    {
+        key: 'slot12',
+        value: 1100,
+        text: '11am - 12pm'
+    },
+    {
+        key: 'slot13',
+        value: 1200,
+        text: '12p - 1pm'
+    },
+    {
+        key: 'slot14',
+        value: 1300,
+        text: '1pm - 2pm'
+    },
+    {
+        key: 'slot15',
+        value: 1400,
+        text: '2pm - 3pm'
+    },
+    {
+        key: 'slot16',
+        value: 1500,
+        text: '3pm - 4pm'
+    },
+    {
+        key: 'slot17',
+        value: 1600,
+        text: '4pm - 5pm'
+    },
+    {
+        key: 'slot18',
+        value: 1700,
+        text: '5pm - 6pm'
+    },
+    {
+        key: 'slot19',
+        value: 1800,
+        text: '6pm - 7pm'
+    },
+    {
+        key: 'slot20',
+        value: 1900,
+        text: '7pm - 8pm'
+    },
+    {
+        key: 'slot21',
+        value: 2000,
+        text: '8pm - 9pm'
+    },
+    {
+        key: 'slot22',
+        value: 2100,
+        text: '9pm - 10pm'
+    },
+    {
+        key: 'slot23',
+        value: 2200,
+        text: '10pm - 11pm'
+    },
+    {
+        key: 'slot24',
+        value: 2300,
+        text: '11pm - 12am'
+    },
+]
+export default class TimePreferencesModal extends Component {
+    constructor(props){
+        super(props)
+        this.state = {
+            day: '',
+            selectedTimes: this.props.timePreferences || [],
+            uncommittedTimesForCurrentDay: [],
+            submitting: false
+        }
+        this.handleDayChange = this.handleDayChange.bind(this)
+        this.getSelectedTimes = this.getSelectedTimes.bind(this)
+        this.removeSelectedTime = this.removeSelectedTime.bind(this)
+        this.handleTimeSlotChange = this.handleTimeSlotChange.bind(this)
+        this.commitAvailability = this.commitAvailability.bind(this)
+        this.submitPreferences = this.submitPreferences.bind(this)
+        this.hasNewPreferences = this.hasNewPreferences.bind(this)
+    }
+
+    hasNewPreferences() {
```

Comment:
Submit button is only enabled when alumnus has added entries that are different from one passed as props, i.e. there are changes to time preferences.